code_annotations/extensions/base.py

```diff
@@ -60,24 +60,9 @@ class SimpleRegexAnnotationExtension(AnnotationExtension):
     Returns a 2-tuple of:
      - ("Comment text", None) in the case of a multi-line comment OR
      - (None, "Comment text") in the case of a single-line comment
```

Comment:
This could certainly be improved. The tuple result is due to the fact that we use `re.match(...).groups()` and not `...groupdict()`. With named groups, we could have a single dict with `singleline_comment` and `multiline_comment` keys.